### PR TITLE
feature/SOF 7395

### PR DESCRIPTION
**Pull request by VsevolodX**
- feat: add defect pair config (2 point defects configs)
- feat: add defect pair builder (applies existing builders twice)

### CODE REVIEW

File: src/py/mat3ra/made/tools/build/defect/__init__.py [hunk 1]
```diff
@@ -13,15 +12,7 @@
...
-from .enums import PointDefectTypeEnum
-
-
-class DefectBuilderFactory(BaseFactory):
-    __class_registry__ = {
-        PointDefectTypeEnum.VACANCY: "mat3ra.made.tools.build.defect.builders.VacancyPointDefectBuilder",
-        PointDefectTypeEnum.SUBSTITUTION: "mat3ra.made.tools.build.defect.builders.SubstitutionPointDefectBuilder",
-        PointDefectTypeEnum.INTERSTITIAL: "mat3ra.made.tools.build.defect.builders.InterstitialPointDefectBuilder",
-    }
+from .enums import PointDefectTypeEnum, DefectBuilderFactory
```
Comment (timurbazhirov) [inline src/py/mat3ra/made/tools/build/defect/__init__.py:15]: Factory is not an enum, let's keep it in this file

File: src/py/mat3ra/made/tools/build/defect/builders.py [hunk 2]
```diff
@@ -322,6 +325,69 @@ def create_adatom(
         return [self.merge_slab_and_defect(new_material, only_adatom_material)]
 
 
+class PointDefectPairBuilder(PointDefectBuilder):
+    _ConfigurationType: type(PointDefectPairConfiguration) = PointDefectPairConfiguration  # type: ignore
+    _GeneratedItemType: Material = Material
+
+    def _create_defect(self, defect_configuration: PointDefectConfiguration) -> Material:
```
Comment (timurbazhirov) [inline src/py/mat3ra/made/tools/build/defect/builders.py:332]: _create_point_defect?

Don't we have a user-facing helper function for this functionality already?

File: src/py/mat3ra/made/tools/build/defect/enums.py [hunk 3]
```diff
@@ -1,13 +1,34 @@
...
 
+from mat3ra.utils.factory import BaseFactory
+
 
 class PointDefectTypeEnum(str, Enum):
     VACANCY = "vacancy"
     SUBSTITUTION = "substitution"
     INTERSTITIAL = "interstitial"
     ADATOM = "adatom"
+    PAIR = "pair"
```
Comment (timurbazhirov) [inline src/py/mat3ra/made/tools/build/defect/enums.py:11]: Should be separate

File: src/py/mat3ra/made/tools/build/defect/enums.py [hunk 4]
```diff
@@ -1,13 +1,34 @@
...
 
 class SlabDefectTypeEnum(str, Enum):
     ISLAND = "island"
     TERRACE = "terrace"
+
+
+class AtomPlacementMethodEnum(str, Enum):
+    COORDINATE = "coordinate"
+    CLOSEST_SITE = "closest_site"
+    EQUIDISTANT = "equidistant"
```
Comment (timurbazhirov) [inline src/py/mat3ra/made/tools/build/defect/enums.py:22]: Let's add comments to explain

File: tests/py/unit/test_tools_build_defect.py [hunk 5]
```diff
@@ -5,27 +5,47 @@
...
     vacuum=6,
     xy_supercell_matrix=[[1, 0], [0, 1]],
     use_orthogonal_z=True,
 )
-t = get_terminations(slab_config)[0]
-slab = create_slab(slab_config, t)
+t_111 = get_terminations(slab_111_config)[0]
+slab_111 = create_slab(slab_111_config, t_111)
+
+slab_001_config = SlabConfiguration(
```
Comment (timurbazhirov) [inline tests/py/unit/test_tools_build_defect.py:39]: Let's put in fixtures to reuse

File: src/py/mat3ra/made/tools/build/defect/builders.py [hunk 6]
```diff
@@ -322,6 +325,69 @@ def create_adatom(
         return [self.merge_slab_and_defect(new_material, only_adatom_material)]
 
 
+class PointDefectPairBuilder(PointDefectBuilder):
```
Comment (timurbazhirov) [inline src/py/mat3ra/made/tools/build/defect/builders.py:328]:
```py

class DefectPairBuilder
    create_defect_pair

class PointDefectPairBuilder(PointDefectBuilder, DefectPairBuilder):
```

File: src/py/mat3ra/made/tools/build/defect/configuration.py [hunk 7]
```diff
@@ -200,3 +203,28 @@ def _json(self):
...
         }
+
+
+class PointDefectPairConfiguration(BaseDefectConfiguration, InMemoryEntity):
+    """
+    Configuration for a pair of point defects.
+
+    Args:
+        primary_defect_configuration (Union[PointDefectConfiguration, AdatomSlabPointDefectConfiguration]):
+        The first defect.
```
Comment (timurbazhirov) [inline src/py/mat3ra/made/tools/build/defect/configuration.py:214]: Type can be elsewhere

File: src/py/mat3ra/made/tools/build/defect/__init__.py (hunk 1 above)
Comment (VsevolodX) [reply src/py/mat3ra/made/tools/build/defect/__init__.py:15]: It's used in ./builders which is imported here, we'll have circular dependency, that's why enums were separated in the first place. The "factory" file will be only solution

**Pull request merged**